estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,83 @@
+from odoo import api, fields, models
+from odoo.tools.date_utils import date
+from odoo.exceptions import UserError, ValidationError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Real Estate Property Offer"
+    _order = "price desc"
+    _sql_constraints = [
+        (
+            "price",
+            "CHECK(price > 0)",
+            "The price must be strictly positive.",
+        )
+    ]
+
+    price = fields.Float()
+    status = fields.Selection(
+        copy=False, selection=[("accepted", "Accepted"), ("refused", "Refused")]
+    )
+    partner_id = fields.Many2one(comodel_name="res.partner", required=True)
+    property_id = fields.Many2one(comodel_name="estate.property", required=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline", inverse="_inverse_date_deadline"
+    )
+    property_type_id = fields.Many2one(
+        comodel_name="estate.property.type",
+        related="property_id.property_type_id",
+        store=True,
+    )
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            create_date = record.create_date or date.today()
+            record.date_deadline = fields.Datetime.add(
+                create_date, days=record.validity
+            )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        all_offers = self.search([])
+        for vals in vals_list:
+            self.env["estate.property"].browse(
+                vals["property_id"]
+            ).state = "offer_received"
```

Comment:
`browse()` is unreliable and should always be suffixed by `exists()`
The point of this method is to create a recordset with some ids inside but it does not verify their existence
```suggestion
            self.env["estate.property"].browse(
                vals["property_id"]
            ).exists().state = "offer_received"
```

Reply:
It could be mentioned in the tutorial ?

Reply:
"Yes indeed, I'll make sure to mention it in the new tutorial" -ANV